openwisp_users/api/mixins.py

```diff
@@ -127,27 +127,28 @@ def _user_attr(self):
     def filter_fields(self):
         try:
             user = self.context['request'].user
-            if user.is_superuser:
-                return
-            organization_filter = getattr(user, self._user_attr)
-            for field in self.fields:
-                if field == 'organization':
-                    self.fields[field].allow_null = False
-                    self.fields[field].queryset = self.fields[field].queryset.filter(
-                        pk__in=organization_filter
-                    )
-                    continue
-                try:
-                    conditions = Q(**{self.organization_lookup: organization_filter})
-                    if self.include_shared:
-                        conditions |= Q(organization__isnull=True)
-                    self.fields[field].queryset = self.fields[field].queryset.filter(
-                        conditions
-                    )
-                except AttributeError:
-                    pass
         except KeyError:
-            pass
+            return
+
+        if user.is_superuser:
+            return
+        organization_filter = getattr(user, self._user_attr)
+        for field in self.fields:
+            if field == 'organization':
+                self.fields[field].allow_null = False
+                self.fields[field].queryset = self.fields[field].queryset.filter(
+                    pk__in=organization_filter
+                )
+                continue
+            try:
+                conditions = Q(**{self.organization_lookup: organization_filter})
+                if self.include_shared:
+                    conditions |= Q(organization__isnull=True)
+                self.fields[field].queryset = self.fields[field].queryset.filter(
+                    conditions
+                )
```

Comment:
There is a `try` block here too, do all the lines need to be inside this block?

Reply:
@atb00ker the error is raised in by line 4
``` python
             conditions = Q(**{self.organization_lookup: organization_filter})
             if self.include_shared:                    
             conditions |= Q(organization__isnull=True)                
             self.fields[field].queryset = self.fields[field].queryset.filter(conditions)
```
Should we only that line in the try statement?

Reply:
@ManishShah120 The question here https://github.com/openwisp/openwisp-users/pull/260#discussion_r650861223
Is repeated to ask the same thing answered here: https://github.com/openwisp/openwisp-users/pull/260#discussion_r650404300

So could you please elaborate the source of the confusion so that I don't give the same answer again! :smile:

Reply:
Done @atb00ker :smile: